### PR TITLE
#9 - Email Collection

### PR DESCRIPTION
**Pull request by autumnation**
Issue #9 

TL;DR - What does this implement?
---------------------------------
Satisfies the requirements of #9, implements a simple method for email collection into a table.
How to Test
-----------
Make a POST request to `http://localhost:5000/add-address` with the following JSON:
```
{
   "email": "test@example.com"
}
```
the first response received should be `action: "created"` and the second `error: "User already exists."`

Worries / Concerns
------------------
I was unsure if and how usernames would work, if they would be used at all, so I omitted having a column for a user's first name or username - if it's necessary to use the user's name, we can loop through the emails in the list and compare them against the database of users, which would reflect accurate information if usernames or user information were to be changed.

Checklist
---------

 - [ ] Do we need to test this?
 - [ ] Were tests written?
 - [x] Does this meet coding standards?
 - [x] Is this is code change?
 - [ ] Is this a meta change (e.g issue template)?
 - [x] Is this pull request ready for review?
 - [ ] Is this a breaking change (e.g not backwards compatible with clients)?
 
ESLint Output
-------------

```
~/Desktop/auth-api >>> eslint ./                                                            
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting.
```

Reviewers
---------

(Tag people here)
@denis-onder 
@gmemstr

### CODE REVIEW

File: app/controller.js [hunk 1]
```diff
@@ -87,6 +88,21 @@ class Controller {
     }
   }
 
+  async addAddress(req, res) {
+    // Add an email address to the database.
+    try {
+      const { email } = req.body;
+      const exists = await Address.findOne({ where: { email } });
+      if (!email) return res.status(500).json({ error: 'No email provided.' });
+      if (exists) return res.status(418).json({ error: 'User already exists.' });
```
Comment (gmemstr) [inline app/controller.js]: I know you want to use this status code, but it's not necessarily the right one for the job here :)

File: app/db/models/add-address.model.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const Sequelize = require('sequelize');
+const database = require('../');
+
+const Address = database.define('Address', {
```
Comment (gmemstr) [inline app/db/models/add-address.model.js]: Bikeshed moment? But `Address` makes me think of a mailing address. Maybe we should make this more specific.

File: app/controller.js (hunk 1 above)
Comment (denis-onder) [reply app/controller.js]: :smile:

File: app/db/models/add-address.model.js (hunk 2 above)
Comment (denis-onder) [reply app/db/models/add-address.model.js]: I agree with gmem, a different model name would be appreciated.

Comment (denis-onder) [inline app/db/models/add-address.model.js]: Rename the model to something else rather than Address

File: app/controller.js [hunk 3]
```diff
@@ -87,6 +88,21 @@ class Controller {
     }
   }
 
+  async addAddress(req, res) {
```
Comment (denis-onder) [inline app/controller.js]: If possible, rename the method to something more "descriptive". Otherwise, code looks fine.
